src/algebra/star/algebra.lean

```diff
@@ -15,31 +16,53 @@ Introduces the notion of a star algebra over a star ring.
 
 universes u v w
 
+open opposite
+
 /--
 A star algebra `A` over a star ring `R` is an algebra which is a star ring,
 and the two star structures are compatible in the sense
 `star (r • a) = star r • star a`.
 -/
--- Note that we take `star_ring A` as a typeclass argument, rather than extending it,
+-- Note that we take `star_monoid A` as a typeclass argument, rather than extending it,
 -- to avoid having multiple definitions of the star operation.
-class star_algebra (R : Type u) (A : Type v)
-  [comm_semiring R] [star_ring R] [semiring A] [star_ring A] [algebra R A] :=
+class star_algebra (R : Type u) (A : Type v) [has_star R] [has_star A] [has_scalar R A] :=
 (star_smul : ∀ (r : R) (a : A), star (r • a) = (@has_star.star R _ r) • star a)
 
 @[simp] lemma star_smul {R : Type u} {A : Type v}
-  [comm_semiring R] [star_ring R] [semiring A] [star_ring A] [algebra R A] [star_algebra R A]
+  [has_star R] [has_star A] [has_scalar R A] [star_algebra R A]
   (r : R) (a : A) :
   star (r • a) = star r • star a :=
 star_algebra.star_smul r a
 
+-- this doesn't work:
+/--
+instance star_ring_equiv.ring_hom_inv_pair {R : Type*} [semiring R] [star_ring R] :
+  ring_hom_inv_pair (↑(star_ring_equiv : R ≃+* Rᵒᵖ) : R →+* Rᵒᵖ) ↑star_ring_equiv.symm :=
+ring_hom_inv_pair.of_ring_equiv _
+-/
+
+-- this does
+attribute [instance] ring_hom_inv_pair.of_ring_equiv
+
+/-- If `A` is a left- and right- module over `R` with compatible actions, then `star` is a
+semilinear equivalence. -/
+@[simps]
+def star_linear_equiv {R : Type u} {A : Type v}
+  [semiring R] [star_ring R] [semiring A] [star_ring A]
+  [module R A] [star_algebra R A]
+  [module Rᵒᵖ A] (h : ∀ (r : R) (a : A), r • a = op r • a) :
+    A ≃ₛₗ[(↑(star_ring_equiv : R ≃+* Rᵒᵖ) : R →+* Rᵒᵖ)] A :=
```

Comment:
If I switch between the `instance` and `attribute [instance]`, lean can't find the typeclass instance it needs for this equiv.